src/Bicep.Core/TypeSystem/Providers/ResourceTypesProviderDescriptor.cs

```diff
@@ -1,33 +1,47 @@
 // Copyright (c) Microsoft Corporation.
 // Licensed under the MIT License.
 
+using Bicep.Core.Registry;
+using Bicep.Core.Semantics.Namespaces;
+
 namespace Bicep.Core.TypeSystem.Providers;
 
-public record ResourceTypesProviderDescriptor
+public class ResourceTypesProviderDescriptor
 {
+    public const string LegacyVersionPlaceholder = "1.0.0";
+
     public ResourceTypesProviderDescriptor(
-        string name,
-        string version,
-        bool isImplicitImport,
-        string? alias = null,
-        string? artifactRef = null,
-        Uri? typesBaseUri = null)
+        string NamespaceIdentifier,
+        Uri ParentModuleUri,
+        string Version = LegacyVersionPlaceholder,
+        string? Alias = null,
+        ArtifactReference? ArtifactReference = null,
+        Uri? TypesDataUri = null)
```

Comment:
I would recommend avoiding overdoing it with default arguments. A pattern you could follow instead to avoid repetition is using static methods - e.g.:

```csharp
public class Foo {
    public Foo(A? valA, B? valB) { ... }

    public static CreateWithA(A valA)
        => new Foo(valA, null);

    public static CreateWithB(B valB)
        => new Foo(null, valB);
}
```

Reply:
When you do use defaults, consider whether it's better to avoid optional params and make people think about the parameter explicitly even if that means having to pass in null.  Not saying that's the case here, just a note.  Another model to consider is fluent interfaces.